docs/reference/extensions/django-framework-extension.rst

```diff
@@ -0,0 +1,292 @@
+- ``<integration>_DB_NAME``
+
+Here, ``<integration>`` is replaced by ``POSTGRESQL``, ``MYSQL``
+``MONGODB`` or ``REDIS`` for the relevant integration. The key
+``optional`` with value ``False`` means that the charm will get blocked
+and stop the services if the integration is not provided.
+
+The provided SAML environment variables are as follows:
+
+- ``SAML_ENTITY_ID`` (required)
+- ``SAML_METADATA_URL`` (required)
+- ``SAML_SINGLE_SIGN_ON_REDIRECT_URL`` (required)
+- ``SAML_SIGNING_CERTIFICATE`` (required)
+
+The S3 integration creates the following environment variables that you
+may use to configure your Flask application:
+
+- ``S3_ACCESS_KEY`` (required)
+- ``S3_SECRET_KEY`` (required)
+- ``S3_BUCKET`` (required)
+- ``S3_REGION``
+- ``S3_STORAGE_CLASS``
+- ``S3_ENDPOINT``
+- ``S3_PATH``
+- ``S3_API_VERSION``
+- ``S3_URI_STYLE``
+- ``S3_ADDRESSING_STYLE``
+- ``S3_ATTRIBUTES``
+- ``S3_TLS_CA_CHAIN``
+
+The RabbitMQ integration creates the connection string in the
+environment variable ``RABBITMQ_CONNECT_STRING``. Furthermore, the
+following environment variables may be provided, derived from the
+connection string:
+
+- ``RABBITMQ_SCHEME``
+- ``RABBITMQ_NETLOC``
+- ``RABBITMQ_PATH``
+- ``RABBITMQ_PARAMS``
+- ``RABBITMQ_QUERY``
+- ``RABBITMQ_FRAGMENT``
+- ``RABBITMQ_USERNAME``
+- ``RABBITMQ_PASSWORD``
+- ``RABBITMQ_HOSTNAME``
+- ``RABBITMQ_PORT``
+- ``RABBITMQ_VHOST``
+
+The environment variable ``DJANGO_BASE_URL`` provides the Ingress URL
+for an Ingress integration or the Kubernetes service URL if there is no
+Ingress integration.
+
+
+HTTP Proxy
+----------
+
+Proxy settings should be set as model configurations. Charms generated
+using the ``django-framework`` extension will make the Juju proxy
+settings available as the ``HTTP_PROXY``, ``HTTPS_PROXY`` and
+``NO_PROXY`` environment variables. For example, the ``juju-http-proxy``
+environment variable will be exposed as ``HTTP_PROXY`` to the Django
+service.
+
+    See more: `Juju | List of model configuration
+    keys <https://juju.is/docs/juju/list-of-model-configuration-keys>`_
+
+
+Background Tasks
+----------------
+
+Extra services defined in the file
+:external+rockcraft:ref:`rockcraft.yaml <rockcraft.yaml_reference>`
+with names ending in ``-worker`` or ``-scheduler`` will be passed the
+same environment variables as the main application. If there is more
+than one unit in the application, the services with the name ending in
+``-worker`` will run in all units. The services with name ending in
+``-scheduler`` will only run in one of the units of the application.
+
+
+Secrets
+-------
+
+Juju secrets can be passed as environment variables to your Django
+application. The secret ID has to be passed to the application as a
+config option in the file ``charmcraft.yaml`` file of type ``secret``.
+This config option has to be populated with the secret ID, in the format
+``secret:<secret ID>``.
+
+The environment variable name passed to the application will be:
+
+.. code-block:: bash
```

Comment:
`text` provides no highlighting, so it's probably more appropriate than `bash`. This isn't quite bash-compatible text, just a pseudocode name.

Strangely, an empty language directive (which we would use in Markdown) defaults to _some_ amount of highlighting, but I can't tell what rules it's using. Sphinx just says it's "default". Not sure what that means.